### DISCUSSION TITLE
do I have to first convert longitude from [0, 360] to [-180, 180]

### DISCUSSION THREAD

**Discussion by shuai-zhou** [corteva/rioxarray#445] [Q&A] [answered]
Still, following @snowman2 recommendation [here](https://gis.stackexchange.com/questions/354782/masking-netcdf-time-series-data-from-shapefile-using-python) to clip a `.nc` file using a shapefile. The longitude of my `.nc` file ranges from **0 to 360** (see below, I made the guess from the `array` of the lon coordinate), but the range of longitude from my shapefile is **-180 to 180**. Do I have to first convert the `.nc` file's longitude from [0, 360] to [-180, 180] through `ds = ds.assign_coords(longitude=(((ds.longitude + 180) % 360) - 180)).sortby('longitude')`. 

In the first place, I thought `ds.rio.write_crs("epsg:4326", inplace=True)` will properly handle this, but it does not change the values of the longitude of the `.nc` file.

![Snipaste_2021-12-30_20-47-17](https://user-images.githubusercontent.com/22311821/147799378-d3501cd3-eb42-4248-bf2e-e6f43a047665.png)

**Comment by snowman2** [comment, marked as answer]
See: https://github.com/corteva/rioxarray/issues/58

You should shift your coordinates to [-180, 180]

**Reply by shuai-zhou** [reply]
Thank you so much for your prompt response. So I have to make sure the longitude's range is [-180, 180] before the clipping?

Two other related questions. 1. Do I also have to check if the latitude to make sure the range is [-90, 90] before clipping (or maybe all the `.nc` files' latitude ranges from [-90, 90] by default?); 2. Do I have to sort the latitude and longitude before clipping? Thanks.

**Reply by snowman2** [reply]
You only need to change the longitude values. I recommend sorting the values, though it may not always be required.

Related: https://gis.stackexchange.com/questions/357246/clip-global-data-by-polygon-using-rioxarray-fails-off-by-180-longitude

**Reply by shuai-zhou** [reply]
Thanks for the answers.
